src/shared/foundation-shared-services/composables/useOpenTab.ts

```diff
@@ -0,0 +1,38 @@
+import { type RouteLocation, useRouter } from "vue-router";
+
+import { useAppHost } from "@dative-gpi/foundation-shared-services/composables";
+
+export const useOpenTab = () => {
+  const { host } = useAppHost();
+  const router = useRouter();
+
+  const handleOpenTabEvent = (event: MouseEvent, target: string | RouteLocation): void => {
+    event.preventDefault();
+
+    // If a new tab is requested, check if it comes from an extension and act accordingly
+    if (event.ctrlKey || event.metaKey || event.button === 1) {
+      openTab(target);
+    }
+    else {
+      router.push(target);
```

Comment:
Le composable s'appel useOpenTab, la méthode handleOpenTabEvent mais en fait dans 90% des cas ça fait juste une navigation classique ? Le retour link fait pas déjà la navigation avec le :to ?

Reply:
J'ai renommé. Si la navigation d'un router-link fait le to par défaut s'il n'y a pas de preventDefault, mais le v-data-iterator n'a pas d'itemTo et vu qu'on l'utilise aussi et que j'ai voulu faire une seule fonction, j'ai mis l'équivalent du comportement par défaut dans tous les cas dans le composable. J'ai remplacé par un booléen qui indique s'il faut faire le router.push pour les v-data-iterator, qui du coup n'est plus utilisé dans le FSRouterLink